framework/src/timesteppers/Stepper.C

```diff
@@ -0,0 +1,748 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "Stepper.h"
+
+StepperInfo::StepperInfo()
+  : _step_count(1),
+    _time(0),
+    _dt(),
+    _nonlin_iters(0),
+    _lin_iters(0),
+    _converged(),
+    _solve_time_secs(),
+    _soln_nonlin(nullptr),
+    _soln_aux(nullptr),
+    _soln_predicted(nullptr),
+    _snapshot(false),
+    _rewind(false),
+    _rewind_time(-1),
+    _dummy_comm()
+{
+  for (int i = 0; i < 3; i++)
```

Comment:
What's the significance of `3` here? Can you use a constant instead? Also, your type should probably be unsigned.